lib/m3u8-parser.js

```diff
@@ -22,6 +27,26 @@ var m3u8parser = module.exports = function() {
       self.emit('item', line.trim());
     }
   }
+  function ytdlParse() {
+    var time = self.ytdlSavedLines[self.ytdlSavedLines.length - 2];
+    if (!time) {
+      self.emit('ytdl-error', new Error('Time is null/undefined.\n' + time));
+      return;
+    }
+    time = parseInt(time.replace(/#EXTINF:/g, "").replace(/,/g, ""));
+    if (self.isFirstTime) time = 1;
+    var line = self.ytdlSavedLines[self.ytdlSavedLines.length-1];
+    if (!line) {
+      self.emit('ytdl-error', new Error('Line is null/undefined.\n' + time));
+      return;
+    }
+    parseLine(line);
+    var timeout = setTimeout(function() {
+      self.ytdlSavedLines = [];
```

Comment:
Reset the array that way when the next packet comes in we're not filling the array with the same useless junk as before. Keeps the array updated and makes sure that no old information is kept.